jetbrains-ultimate/src/software/aws/toolkits/jetbrains/services/lambda/go/GoSamProjectWizard.kt

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.go
+
+import com.goide.sdk.GoSdkService
+import com.goide.sdk.combobox.GoSdkChooserCombo
+import com.goide.vgo.configuration.VgoProjectSettings
+import com.intellij.facet.ui.ValidationResult
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.roots.ModifiableRootModel
+import com.intellij.openapi.ui.TextFieldWithBrowseButton
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.openapi.vfs.VirtualFile
+import software.amazon.awssdk.services.lambda.model.PackageType
+import software.amazon.awssdk.services.lambda.model.Runtime
+import software.aws.toolkits.jetbrains.services.lambda.wizard.SamAppTemplateBased
+import software.aws.toolkits.jetbrains.services.lambda.wizard.SamNewProjectSettings
+import software.aws.toolkits.jetbrains.services.lambda.wizard.SamProjectTemplate
+import software.aws.toolkits.jetbrains.services.lambda.wizard.SamProjectWizard
+import software.aws.toolkits.jetbrains.services.lambda.wizard.SdkSelector
+import software.aws.toolkits.jetbrains.utils.ui.validationInfo
+import software.aws.toolkits.resources.message
+import javax.swing.JComponent
+import javax.swing.JLabel
+
+class GoSamProjectWizard : SamProjectWizard {
+    override fun createSdkSelectionPanel(projectLocation: TextFieldWithBrowseButton?): SdkSelector = GoSdkSelectionPanel()
+
+    override fun listTemplates(): Collection<SamProjectTemplate> = listOf(
+        SamHelloWorldGo()
+    )
+}
+
+class GoSdkSelectionPanel : SdkSelector {
+    private val interpreterPanel = GoSdkChooserCombo()
+
+    override fun sdkSelectionLabel() = JLabel(message("sam.init.go.sdk"))
+
+    override fun sdkSelectionPanel(): JComponent = interpreterPanel
+
+    override fun validateSelection(): ValidationInfo? = interpreterPanel.validator.validate(interpreterPanel.sdk)?.let {
+        if (it == ValidationResult.OK) {
+            return null
+        }
+        interpreterPanel.validationInfo(it.errorMessage)
+    }
+
+    override fun applySdkSettings(model: ModifiableRootModel) {
+        GoSdkService.getInstance(model.project).setSdk(interpreterPanel.sdk)
+    }
+}
+
+class SamHelloWorldGo : SamAppTemplateBased() {
+    override fun postCreationAction(settings: SamNewProjectSettings, contentRoot: VirtualFile, rootModel: ModifiableRootModel, indicator: ProgressIndicator) {
+        // Turn on vgo integration, required for it to resolve dependencies properly
+        VgoProjectSettings.getInstance(rootModel.project).isIntegrationEnabled = true
```

Comment:
maybe the is needs to be a pre-action instead (before sam init spits out files)? iirc you were getting a popup asking if you wanted to enable go-mod support, even though it was already on

Reply:
Hm that didn't fix it (and actually broke it by making the setting not work), tried some other settings in there but there doesn't seem to be a way to make it not pop up without setting the IDE setting to never show it (which sounds worse)